backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -67,35 +72,35 @@ public void sendVerificationEmail(String emailAddress) {
       throw new UserException(UserExceptionType.USER_ALREADY_REGISTERED);
     }
 
-    String token = emailUtils.generateJwtEmailToken(emailAddress);
+    String token = JwtUtils.generateEmailToken(emailAddress, JwtUtils.EMAIL_VERIFICATION_PURPOSE);
+
+    Map<String, String> arguments =
+        Map.of("emailAddress", emailAddress, "link", UrlUtil.getEmailVerificationLink(token));
```

Comment:
why do you need the emailAddress property in the map, when you propagate it as separate field in Email constructor? Btw you can always extract Email creation in separate method for logic separation and better readability.. even new EmailHelper class is perfectly fine if it makes sense to you

Reply:
Forgot to change it, I intended to send the Users name in the arguments, sounds weird reffering to them by their email.

Reply:
@petarstoimenovskihtec  Do you mean something like createVerificationEmail() in some sort of helper or util class?

Reply:
yup, something like that